src/System.Management.Automation/engine/parser/TypeResolver.cs

```diff
@@ -690,7 +690,8 @@ internal static class CoreTypes
                     { typeof(PSCredential),                                new[] { "pscredential" } },
                     { typeof(PSDefaultValueAttribute),                     new[] { "PSDefaultValue" } },
                     { typeof(PSListModifier),                              new[] { "pslistmodifier" } },
-                    { typeof(PSObject),                                    new[] { "psobject", "pscustomobject" } },
+                    { typeof(PSObject),                                    new[] { "psobject" } },
+                    { typeof(PSCustomObject),                              new[] { "pscustomobject" } },
```

Comment:
I could be wrong, but it seems inconsistent with the current proposed conversion: the acelerator is for type `PSCustomObject`, but binder for hashtable does the conversion to `PSObject`. One option would be to completely eliminate `PSCustomObject` and set type accelerator to  `PSObject`, if that doesn't break anything.